### PR TITLE
informing user of dead session

### PR DESCRIPTION
**Pull request by karina-y**
- letting user know when their session has been closed due to inactivity
- had to refactor logpoints a bit due to a circular reference from socket.ts, only setting up log points now if we have a recording
- force testing this works fine but i'm going to leave a window open tonight and check it in the morning to confirm

### CODE REVIEW

File: src/protocol/logpoint.ts [hunk 1]
```diff
@@ -55,75 +55,77 @@ export const PointHandlers: {
   onPoints?: (points: PointDescription[], info: LogpointInfo) => void;
 } = {};
 
-client.Analysis.addAnalysisResultListener(({ analysisId, results }) => {
-  log(`AnalysisResults ${results.length}`);
+export function setupLogpoints() {
```
Comment (jasonLaster) [inline src/protocol/logpoint.ts:58]: why did this need to be in a setup function?

could we move the handler bits out `({ analysisId, results }) => {})`

so it reads

```js
export function setupLogpoints() {
  client.Analysis.addAnalysisResultListener((onAnalysisResult))
}
```

File: src/protocol/socket.ts [hunk 2]
```diff
@@ -113,6 +114,12 @@ function onSocketMessage(evt: MessageEvent<any>) {
 function onSocketClose() {
   setStatus("Disconnected.");
   log("Socket Closed");
+
+  store.dispatch(
```
Comment (jasonLaster) [inline src/protocol/socket.ts:118]: is the store global? Is there any way to pass it in... 

perhaps ... https://gist.github.com/73b610bde300ea6e50c5ef58ad18ac90

Comment (karina-y) [reply src/protocol/socket.ts:118]: it is global and it does work when i test so i'm not sure why it's throwing an error on store. but i'll add your changes in and see if it fixes that error

File: src/protocol/logpoint.ts [hunk 3]
```diff
@@ -123,7 +124,13 @@ client.Analysis.addAnalysisPointsListener(({ analysisId, points }) => {
       LogpointHandlers.onPointLoading!(logGroupId, point, time, location);
     });
   }
-});
+};
+
+export function setupLogpoints() {
+  client.Analysis.addAnalysisResultListener(({ analysisId, results }) => onAnalysisResult);
```
Comment (jasonLaster) [inline src/protocol/logpoint.ts:130]: shouldn't this be?

```js
client.Analysis.addAnalysisResultListener(({ analysisId, results }) => onAnalysisResult(analysisId, results))
```

Comment (karina-y) [reply src/protocol/logpoint.ts:130]: you're absolutely right, i'll start leaving late changes for the morning so i don't miss things like this.

**Comment by karina-y** [comment]
inspector-02 is timing out in ci, local branch, and master

**Pull request merged**